src/pyflex/window_selector.py

```diff
@@ -316,25 +339,28 @@ def select_windows(self):
 
         self.calculate_preliminiaries()
 
+        self.determine_signal_and_noise_indices()
+        if self.config.check_global_data_quality:
+            if not self.check_data_quality():
+                return []
+
```

Comment:
I think I understood...

One thing, I feel like we need another check for noise_end_index(also for noise_start_index and other index) in case people assign unreasonable values to those index.

For example, for a 100 elements array. If people assign noise_start_index = 5 and noise_end_index = 15. It makes sense since noise_end_indx > noise_start_index. But if people set noise_start_idx = 5 and noise_end_index = 2, then it does not make sense.

However, since python allows negative index, I think we need some tricks here to make sure it is robust.

I am talking about extreme cases..but I think it is nice to have those kinds of checks.

If you already have those checks somewhere, just forget what I said...